### DISCUSSION TITLE
Separate out theming into its own pacakge.

### DISCUSSION THREAD

**Discussion by flyingcircle** [react-native-elements/react-native-elements#3268] [Ideas]
Theming in RNE is unique in that I don't know of any other theming package in React Native that handles theming the way we do. Unfortunately however, during the move to typescript, it became clear that reasoning around the types for theming is complicated, and any user of RNE that doesn't want to use themes has an unnecessary added layer of complexity. 

My suggestion here is that for RNE 5.0, we remove theming and extract out into its own pacakge `react-native-elements-themed`. Can anyone think of pros/cons to this?

CC: @Monte9  @pranshuchittora  @arpitBhalla @khushal87 

Or perhaps if we continue to expand the number of packages in RNE we should start scoping the package names: `@react-native-elements/base` `@react-native-elements/themed` etc.

**Comment by arpitBhalla** [comment]
Will `@react-native-elements/themed` be components of `@react-native-elements/base` with a HOC of `withTheme`?

**Comment by flyingcircle** [comment]
Yeah that's my thinking. Basically pull out the default HOC, and have a separate package where that is added to the components.

**Reply by arpitBhalla** [reply]
Can we export un-themed components from another sub directory?

Edit: And this, within the same package

**Reply by flyingcircle** [reply]
Well, part of my hope was to simplify the code a bit. Just exporting the components to another package may be simpler to implement, but it doesn't simplify the code.

**Comment by arpitBhalla** [comment]
### Usage 
```ts
import { Button } from 'react-native-elements';
// For themed component
import { LinearProgress } from 'react-native-elements/themed';
```
If we started using scoped packages
```ts
import { Button } from '@react-native-elements/base';
// For themed component
import { LinearProgress } from '@react-native-elements/base/themed';
```

### Exports

**index.ts**
```ts
export { LinearProgress } from './LinearProgress';
export type { LinearProgressProps } from './LinearProgress';
```

**themed.ts**
With `withTheme` HOC
```ts
export { LinearProgress } from './LinearProgress/LinearProgress.themed';
export type { LinearProgressProps } from './LinearProgress/LinearProgress.themed';
```


### package.json
```json
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "exports": {
    ".": "./dist/index.js",
    "./themed": "./dist/themed.js"
  },
  "typesVersions": {
    "*": {
      "themed": ["dist/themed.d.ts"]
    }
  },
```
